tools/fog-local-network/fog_conformance_tests.py

```diff
@@ -764,6 +765,54 @@ def run(self):
             self.follow_up_balance_check(wallet3, {8: 1}, [8])
             self.follow_up_balance_check(wallet4, {8: 14}, [8])
 
+
+        #######################################################################
+        # Test that when we change the expiry window, we record when was the 
+        # last time the report has been cached.  The desired behavior is that
+        # when the pubkey expiry window is N blocks, then the last
+        # reported published block should be N blocks ago.
+        #######################################################################
+
+        # Expecting index=7
+        index1 = self.fog_ingest.get_last_report_published_block_index()
+        print("===> last report published block index: ", index1)
+
+        # Add nothing
+        # Take 4 from 4
+        credits7x = [{'account': 0, 'amount': 0},
+                     {'account': 1, 'amount': 0},
+                     {'account': 2, 'amount': 0},
+                     {'account': 3, 'amount': 0},
+                     {'account': 4, 'amount': 0}]
+        key_images7x = [block7_key_images[0]]
+        print("===> Key images spent in Block 7x: ", key_images7x)
+        block7x_key_images = ledger1.add_block(credits7x, key_images7x, fog_pubkey)
+        ledger2.add_block(credits7x, key_images7x, fog_pubkey)
+        print("===> Key images for new transactions in Block 7x: ", block7x_key_images)
+        time.sleep(1)
+
+        # Expecting index=8
+        index2 = self.fog_ingest.get_last_report_published_block_index()
+        print("===> last report published block index: ", index2)
+
+        assert (index2 != index1) and (index2 > index1)
+
```

Comment:
Maybe add a step that changes the expiry window to 3, adds 3 blocks, and verifies after each block that the report hasn't been published until it reaches the third block.

@eranrund - can you check our logic here: the intent is for this test to verify that changing the expiry window is properly handled in conjunction with report publishing (particularly when the report is no longer published on every block).

@queenbdubs explained why reducing the pubkey expiry window is problematic - but we do still need to make sure that the code handles this correctly - both on increase and decrease of expiry window.

IIUC, in the wild, we may increase the expiry window for some reason (maybe if transaction volume is high and blocks are being written quickly?), and if we decreased it later because transaction volume settled down, we would still keep the report valid up until the last block in the expiry window, then decrease it after that block.

Reply:
Yes, this seems correct - I believe increasing should work, decreasing should result in an error.